### PR TITLE
HTTP handlers by-ref/by-val/by-ptr semantics + minor changes.

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

**Comment by dkorolev** [comment]
@deathbaba : Please review.
@sompylasar : FYI.

Makefile change also in https://github.com/KnowSheet/KnowSheet/pull/2.

**Comment by dkorolev** [comment]
@deathbaba ping?

File: cerealize/test/impl.cc [hunk 1]
```diff
@@ -202,6 +202,9 @@ TEST(Cerealize, ConsumerSupportsPolymorphicTypes) {
 
     // This function will not be called since the `EventAppSuspend` type is not in `DERIVED_TYPE_LIST` above.
     void operator()(const EventAppSuspend& e) { e.AppendTo(os) << " *** NOT CALLED SUSPEND *** " << std::endl; }
+
+    ExampleConsumer() {}
```
Comment (deathbaba) [inline cerealize/test/impl.cc]: = default?

Comment (dkorolev) [inline cerealize/test/impl.cc]: Fixed.

**Pull request merged**